python/lsst/ts/ess/common/data_client/snmp_data_client.py

```diff
@@ -118,7 +135,14 @@ def __init__(
         # Attributes for the SNMP requests.
         self.snmp_engine = SnmpEngine()
         self.community_data = CommunityData(self.config.snmp_community, mpModel=0)
-        self.transport_target = UdpTransportTarget((self.config.host, self.config.port))
+        if hasattr(UdpTransportTarget, "create"):
```

Comment:
I think this might be a bit confusing, especially because later you have to overwrite `self.transport_target` with an `self.transport_target = await self.transport_target`.

Here is what I would suggest, make `self.transport_target = None` here if you need to create it. Something like

```
self.transport_target = None if hasattr(UdpTransportTarget, "create") else UdpTransportTarget((self.config.host, self.config.port))
```

Then, in `execute_next_cmd_non_blocking` you can do:

```
if self.transport_target is None:
    self.transport_target = await UdpTransportTarget.create((self.config.host, self.config.port))
```

You may have to sprinkle some `assert self.transport_target is not None` here and there to appease mypy if do this, but I think it is work the added clarity.